### PR TITLE
Send user ID and facility ID along with Sentry reports

### PR DESCRIPTION
**Pull request by vinaysshenoy**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt [hunk 1]
```diff
@@ -23,7 +25,29 @@ class SentryCrashReporter(
...
+        .subscribeOn(io())
+        .filterAndUnwrapJust()
+        .replay()
+        .refCount()
+
+    loggedInUserStream
+        .map { it.uuid }
+        .subscribe({
+          Sentry.getContext().addTag("userUuid", it.toString())
+        }, { report(it) })
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: This formatting looks strange. Let's keep the arguments on separate lines?

```kotlin
.subscribe(
    { Sentry.getContext().addTag("userUuid", it.toString()) },
    { report(it) })
```

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt [hunk 2]
```diff
@@ -23,7 +25,29 @@ class SentryCrashReporter(
...
 
+    loggedInUserStream
+        .flatMap { facilityRepository.currentFacility(it) }
+        .map { it.uuid }
+        .subscribe({
+          Sentry.getContext().addTag("facilityUuid", it.toString())
+        }, { report(it) })
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: same comment

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt [hunk 3]
```diff
@@ -23,7 +25,29 @@ class SentryCrashReporter(
 
   override fun init(appContext: Application) {
     Sentry.init(AndroidSentryClientFactory(appContext))
+    keepUserAndFacilityRegistered()
+  }
+
+  @Suppress("CheckResult")
+  private fun keepUserAndFacilityRegistered() {
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: `keepUserAndFacilityRegistered()` -> `identifyUserAndCurrentFacility()`?

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: Done

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: Done

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt]: Done

**Pull request merged**